packages/strapi-admin/services/permission/sections-builder/builder.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const createSection = require('./section');
+
+const createSectionBuilder = () => {
+  const state = {
+    sections: new Map(),
+  };
+
+  return {
+    addSection(sectionName, options) {
+      const section = createSection(options);
```

Comment:
looks like the createSection should be done outside. (like in action provider or condtion provider btw)

you could add createSection to make it clearer you create a section & not add an existing one